catalyst-gateway/bin/src/follower.rs

```diff
@@ -0,0 +1,337 @@
+                    "No previous followers, staleness not relevant. Start follower from genesis."
+                );
+                DATA_NOT_STALE
+            };
+
+            // Threshold which defines if data is stale and ready to update or not
+            if chrono::offset::Utc::now().timestamp() - threshold > configs.1.timing_pattern.into()
+            {
+                info!(
+                    "Last update is stale for network {} - ready to update, starting follower now.",
+                    config.network
+                );
+                let follower_handler = init_follower(
+                    network,
+                    &config.relay,
+                    (slot_no, block_hash),
+                    db.clone(),
+                    machine_id.clone(),
+                    &snapshot_path,
+                )
+                .await?;
+                break follower_handler;
+            }
+            info!(
+                "Data is still fresh for network {}, tick until data is stale",
+                config.network
+            );
+        };
+
+        follower_tasks.push(task_handler);
+    }
+
+    Ok(follower_tasks)
+}
+
+/// Establish point at which the last follower stopped updating in order to pick up where
+/// it left off. If there was no previous follower, start indexing from genesis point.
+async fn find_last_update_point(
+    db: Arc<EventDB>, network: &String,
+) -> Result<(Option<SlotNumber>, Option<BlockHash>, Option<LastUpdate>), Box<dyn Error>> {
+    let (slot_no, block_hash, last_updated) =
+        match db.last_updated_metadata(network.to_string()).await {
+            Ok((slot_no, block_hash, last_updated)) => {
+                info!(
+                "Previous follower stopped updating at Slot_no: {} block_hash:{} last_updated: {}",
+                slot_no, block_hash, last_updated
+            );
+                (Some(slot_no), Some(block_hash), Some(last_updated))
+            },
+            Err(err) => {
+                info!("No previous followers, start from genesis. Db msg: {}", err);
+                (None, None, None)
+            },
+        };
+
+    Ok((slot_no, block_hash, last_updated))
+}
+
+/// Initiate single follower and returns associated task handler
+/// which facilitates future control over spawned threads.
+async fn init_follower(
+    network: Network, relay: &str, start_from: (Option<SlotNumber>, Option<BlockHash>),
+    db: Arc<EventDB>, machine_id: MachineId, snapshot: &str,
+) -> Result<ManageTasks, Box<dyn Error>> {
+    let mut follower = follower_connection(start_from, snapshot, network, relay).await?;
+
+    let genesis_values =
+        network_genesis_values(&network).ok_or("Obtaining genesis values failed")?;
+
+    let task = tokio::spawn(async move {
+        loop {
+            let chain_update = match follower.next().await {
+                Ok(chain_update) => chain_update,
+                Err(err) => {
+                    error!(
```

Comment:
Should this error (and all the other below) be logged as warn/info since we are just skipping it ?

Reply:
This situation shouldn't really happen, but we don't want it to stop the followers either. I think it should be error to increase the visibility. We can change it in the future if its too noisy.